compiler/hie-core/src/Development/IDE/LSP/LanguageServer.hs

```diff
@@ -32,14 +35,18 @@ import Language.Haskell.LSP.Messages
 
 
 runLanguageServer
-    :: ((FromServerMessage -> IO ()) -> VFSHandle -> IO IdeState)
+    :: LSP.Options
+    -> PartialHandlers
+    -> ((FromServerMessage -> IO ()) -> VFSHandle -> IO IdeState)
     -> IO ()
-runLanguageServer getIdeState = do
+runLanguageServer options userHandlers getIdeState = do
     -- Move stdout to another file descriptor and duplicate stderr
     -- to stdout. This guards against stray prints from corrupting the JSON-RPC
     -- message stream.
     newStdout <- hDuplicate stdout
     stderr `hDuplicateTo` stdout
+    hSetBuffering stderr NoBuffering
```

Comment:
Those are the handles that LSP uses for communication on. The purpose of this change is so that our logging messages arrive promptly. The reason for adding these changes is that I was observing huge amounts of buffering, which was making debugging way harder.